### PR TITLE
New Spring Annotations Cheat Sheet

### PR DESCRIPTION
**Pull request by HansTheRisk**
## Description of new Instant Answer, or changes

Displays the most common Spring-Framework annotations.

## People to notify

@pjhampton

---

Instant Answer Page: https://duck.co/ia/view/spring_annotations_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@HansTheRisk Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/spring-annotations.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		"spring-security annotations"
+	],
+    	"section_order": [
+		"Spring Annotations",
+		"Spring MVC Annotations",
+		"Spring Security Annotations"
+    	],
+	"sections": {
+		"Spring Annotations": [{
+			"val": "Annotate all your service classes with @Service.",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/spring-annotations.json]: Lets remove periods from end of val fields

File: share/goodie/cheat_sheets/json/spring-annotations.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
+{
+	"id": "spring_annotations_cheat_sheet",
+	"name": "Spring Annotations",
+	"description": "Overview of common Spring-framework annotations",
+	"template_type": "code",
+	"metadata": {
+		"sourceName": "Spring Annotations",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/spring-annotations.json]: Source name is "TechFerry"

**Comment by tagawa** [comment]
Thanks for this new cheat sheet @HansTheRisk, and for the quick review @jophab.
It's on the beta server for testing: https://beta.duckduckgo.com/?q=spring+framework+annotations+cheatsheet&ia=cheatsheetscontroller&iax=1

File: share/goodie/cheat_sheets/json/spring-annotations.json (hunk 1 above)
Comment (HansTheRisk) [inline share/goodie/cheat_sheets/json/spring-annotations.json]: Got both fixed :+1: Thanks.

**Comment by tagawa** [comment]
@hanstherisk @jophab Thanks again both of you - let's merge this! It should be released on the production server within a week or so.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=spring framework annotations)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/spring_annotations_cheat_sheet_mobile.png?7W1NMHRpDe)

**Pull request merged**